### PR TITLE
Adds global phase to the `PhaseShift` decomposition

### PR DESCRIPTION
**Pull request by obliviateandsurrender**
**Context:** Adds `GlobalPhase` to `PhaseShift` decomposition, which is right now missing and simply expands to `RZ`.

**Description of the Change:** Changes the `compute_decomposition` for `PhaseShift` and a few tests using the previous decomposition

**Benefits:** We wouldn't miss out on the phase information when decomposing this gate. 

**Possible Drawbacks:** N/A

**Related GitHub Issues:** N/A

**Note for reviewers:**
If a device does not support `PhaseShift` and _also_ doesn't support `GlobalPhase`, this should not cause any trouble because `GlobalPhase` decomposes to nothing. It won't have the phase information, but it didn't in the past either (just the `RZ`, so it will remain correct up to a phase)

### CODE REVIEW

File: tests/transforms/test_batch_transform.py [hunk 1]
```diff
@@ -238,9 +240,13 @@ def my_transform(self, tape, **kwargs):
...
         input_tape = spy_transform.call_args[0][1]
-        assert len(input_tape.operations) == 1
-        assert input_tape.operations[0].name == ("RZ" if perform_expansion else "PhaseShift")
+
+        assert len(input_tape.operations) == 2 if perform_expansion else 1
+        assert input_tape.operations[0].name == "RZ" if perform_expansion else "PhaseShift"
         assert input_tape.operations[0].parameters == [0.5]
+        if perform_expansion:
+            assert input_tape.operations[1].name == "GlobalPhase"
+            assert input_tape.operations[1].parameters == [0.25]
```
Comment (Jaybsoni) [inline tests/transforms/test_batch_transform.py:249]: Can put it all under one if condition

```suggestion
        if perform_expansion:
            assert len(input_tape.operations) == 2
            assert input_tape.operations[0].name == "RZ"
            assert input_tape.operations[0].parameters == [0.5]
            assert input_tape.operations[1].name == "GlobalPhase"
            assert input_tape.operations[1].parameters == [0.25]
        else: 
            assert len(input_tape.operations) == 1
            assert input_tape.operations[0].name == "PhaseShift"
            assert input_tape.operations[0].parameters == [0.5]
```

File: tests/transforms/test_batch_transform.py [hunk 2]
```diff
@@ -276,10 +282,15 @@ def qnode(x):
...
-        assert len(input_tape.operations) == 1
-        assert input_tape.operations[0].name == ("RZ" if perform_expansion else "PhaseShift")
+
+        assert len(input_tape.operations) == 2 if perform_expansion else 1
+        assert input_tape.operations[0].name == "RZ" if perform_expansion else "PhaseShift"
         assert input_tape.operations[0].parameters == [0.5]
 
+        if perform_expansion:
+            assert input_tape.operations[1].name == "GlobalPhase"
+            assert input_tape.operations[1].parameters == [0.25]
```
Comment (Jaybsoni) [inline tests/transforms/test_batch_transform.py:292]: See comment above about putting under a single if check

File: pennylane/transforms/optimization/optimization_utils.py [hunk 3]
```diff
@@ -185,3 +197,23 @@ def fuse_rot_angles(angles_1, angles_2):
         return _no_fuse(angles_1, angles_2)
 
     return _fuse(angles_1, angles_2)
+
+
+def _fuse_global_phases(operations):
+    """Fuse all the global phase operations into single one
```
Comment (Jaybsoni) [inline pennylane/transforms/optimization/optimization_utils.py:203]:
```suggestion
    """Fuse all the global phase operations into a single one.
```

**Comment by Jaybsoni** [comment]
Just a few small comments, I think it would also be useful to look for decompositions of PhaseShift that don't use the GlobalPhase operator until we have a decomposition for it

**Comment by trbromley** [comment]
> I'm not sure what would change after 0.33 is released, but we might need to work out a better plan for general GlobalPhase support before doing this.

Agreed, I think we can just give a decomposition to the identity. This is wrong, but only as wrong as it was before.

**Comment by timmysilv** [comment]
works for me... but what wire will we give to the decomposition 😅

**Comment by trbromley** [comment]
> works for me... but what wire will we give to the decomposition 😅

I want to say that it should be the wires of `GlobalPhase`, but I'm still confused by the [docstrings](https://docs.pennylane.ai/en/latest/code/api/pennylane.GlobalPhase.html) about wires being an "unused argument".

**Comment by timmysilv** [comment]
the op has no wires. the argument is only there to not break constructors - Lillian tried to remove it and it made a huge mess for some reason. If you give them and they match all device wires, nice! If they don't match, your circuit will fail. But if you don't give them at all, the op wires are an empty list. So, if you try to decompose to Identity in that case (a common one, who would give wires to a global phase??), we won't know what wires to use.

idea: remove the need for wires from Identity as well.

**Review by timmysilv — APPROVED** [review]
even better, GlobalPhase just decomposes to... nothing! (#4855)

add a changelog entry and this lgtm!

**Review by vincentmr — APPROVED** [review]
Looks good to me, thanks @obliviateandsurrender .

**Pull request merged**